src/pyodide.js

```diff
@@ -291,7 +305,38 @@ globalThis.languagePluginLoader = new Promise((resolve, reject) => {
     if (!Array.isArray(names)) {
       names = [ names ];
     }
+    // get shared library packages and load those first
+    // otherwise bad things happen with linking them in firefox.
+    sharedLibraryNames = [];
+    try {
+      sharedLibraryPackagesToLoad =
+          recursiveDependencies(names, messageCallback, errorCallback, true);
+      for (pkg of sharedLibraryPackagesToLoad) {
+        sharedLibraryNames.push(pkg[0]);
+      }
+    } catch (e) {
+      // do nothing - let the main load throw any errors
+    }
+    let allLibs = [];
+    let oldOpen = Module.FS.open;
+    newOpen = (path, flags, mode) => {
+      allLibs.push(path);
+      return oldOpen(path, flags, mode);
+    };
+    Module.FS.open = newOpen;
     let promise = loadPackageChain.then(
+        () => _loadPackage(sharedLibraryNames, messageCallback || console.log,
+                           errorCallback || console.error));
+    loadPackageChain = loadPackageChain.then(() => promise.catch(() => {}));
+    await promise;
+    Module.FS.open = oldOpen;
+    console.log("LIBS:", allLibs);
```

Comment:
We should probably remove this debugging line, or at least use `messageCallback` if defined.